### ISSUE TITLE
Regex in filterSearch causes error

### ISSUE THREAD

**Issue by chelseyong** [reposense/RepoSense#571]
When a `\` is placed at the end of filterSearch (search bar), it was interpreted as regex and catches this error:

![image](https://user-images.githubusercontent.com/30564248/53300150-baee0180-3875-11e9-8606-c24ac788fd28.png)

**Comment by eugenepeh** [comment]
Since this is not a non-recoverable bug and neither of the searchable parameters allow the use of `\` character, putting it as low priority for now.

**Comment by jylee-git** [comment]
I can't think of any scenario where `\` may be included in the author's name or repository. Should we just ignore the character?

**Comment by eugenepeh** [comment]
>I can't think of any scenario where \ may be included in the author's name or repository. Should we just ignore the character if the escape character is detected?

No, as it uses regex matching, it can be difficult to isolate all the individual `\`,
best solution would probably to ignore the input if its an invalid regex.
However, though unsightly, this issue is not important. It does not cause the browser to stop functioning because of that, hence it can be ignored for now.
